sdk/core/src/main/java/com/klaviyo/coresdk/networking/KlaviyoApiClient.kt

```diff
@@ -0,0 +1,180 @@
+package com.klaviyo.coresdk.networking
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Looper
+import com.klaviyo.coresdk.Registry
+import com.klaviyo.coresdk.lifecycle.ActivityEvent
+import com.klaviyo.coresdk.model.Event
+import com.klaviyo.coresdk.model.Profile
+import com.klaviyo.coresdk.networking.requests.IdentifyApiRequest
+import com.klaviyo.coresdk.networking.requests.KlaviyoApiRequest
+import com.klaviyo.coresdk.networking.requests.TrackApiRequest
+import java.util.concurrent.ConcurrentLinkedQueue
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * Coordinator of API request traffic
+ */
+internal object KlaviyoApiClient : ApiClient {
+    internal const val QUEUE_KEY = "klaviyo_api_request_queue"
+
+    private val handlerThread = HandlerUtil.getHandlerThread(KlaviyoApiClient::class.simpleName)
+    private var handler: Handler? = null
+    private var apiQueue = ConcurrentLinkedQueue<KlaviyoApiRequest>()
```

Comment:
My impression is no, but I haven't gone deep on threading yet. I copied this handler/looper stuff forward from existing code. If we refactor this in the future, I believe co-routines would be the modern way to handle this, but I don't think this would be considered deprecated or even outdated either.

Reply:
Ah I was wrong, there are thread priorities 1-10. Default seems to be fine for us though.